docs/porting/security/psa_targets_in_mbed-os.md

```diff
@@ -0,0 +1,561 @@
+            "MBED_MPU_CUSTOM"
+        ],
+        "extra_labels_add": [
+            "ARM_SSG",
+            "MUSCA_A1",
+            "MUSCA_A1_NS"
+        ],
+        "post_binary_hook": {
+            "function": "ArmMuscaA1Code.binary_hook"
+        },
+        "secure_image_filename": "tfm_s.bin",
+        "tfm_target_name": "MUSCA_A",
+        "tfm_bootloader_supported": true,
+        "tfm_default_toolchain": "ARMCLANG",
+        "tfm_supported_toolchains": [
+            "ARMCLANG",
+            "GNUARM"
+        ],
+        "tfm_delivery_dir": "TARGET_ARM_SSG/TARGET_MUSCA_A1"
+    }
+```
+
+Please note the config options `macros_add`, `extra_labels_add` and `device_has_remove`. To add or remove `macros`, `extra_labels` or target capabilities, a PSA target definition must use
+[macros/extra_labels/device_has]`_add` or [macros/extra_labels/device_has]`_remove` (not `macros`, `extra_labels` or `device_has`).
+
+To add or remove a feature, use `feature_`[add/remove].
+
+By default, a TF-M build generates a `bin` file. If the target requires a `hex` file, you need to add the attribute `"TFM_OUTPUT_EXT": "hex"` to the target definition. The build script will convert `bin` to `hex`, then copy the `hex` to `tfm_delivery_dir`. You must also update `secure_image_filename` to match the new file extension.
+
+If an Armv8-M PSA target cannot inherit from `PSA_V8_M` because it has to inherit from one of its family targets, you must add the following attributes:
+
+```json
+        "features_add": [
+            "PSA"
+        ],
+        "extra_labels_add": [
+            "TFM",
+            "TFM_V8M"
+        ],
+        "macros_add": [
+            "MBEDTLS_PSA_HAS_ITS_IO",
+            "MBEDTLS_USE_PSA_CRYPTO"
+        ]
+```
+
+## Enabling PSA at application level
+
+Having an entropy source is crucial for Mbed TLS and Mbed Crypto. The [entropy source porting document](../porting/entropy-sources.html) discusses why and how to add an entropy source.
+
+If your target doesn't have a True Random Number Generator (TRNG), configure it as a non-PSA target in `targets/targets.json`. In this scenario, if an application wants to use the target as a PAS target, then it is the application's responsibility to provide an entropy source and mark that target as a PSA target at the application level. To enable PSA for a target from an application, use the config option [target_overrides](,,/reference/configuration.html).
+
+An example of `mbed_app.json`:
+
+```json
+"target_overrides": {
+    "K64F": {
+        "inherits": ["PSA_V7_M"]
+    }
+}
+```
+
+## Build and validation
+
+For dual-core and Armv8-M PSA targets, TF-M runs on the SPE and provides PSA services<!--I feel like we've said this at least twice already-->. The Python script `build_tfm.py` automates building TF-M and copying the TF-M binary to the location defined by the target attribute `tfm_delivery_dir`. Another Python script, `build_psa_compliance.py`, automates
```

Comment:
```suggestion
For dual-core and Armv8-M PSA targets, TF-M runs on the SPE and provides PSA services<!--I feel like we've said this at least twice already-->. The Python script `build_tfm.py` automates building TF-M and copying the TF-M binary to the location defined by the target attribute `tfm_delivery_dir`. Another Python script, `build_psa_compliance.py`, automates
```

Reply:
sure, it’s just a pre-text to define (again) SPE and NSPE and then say how to build SPE. Can be removed.